tools/loadtesting/terraform/ecs.tf

```diff
@@ -0,0 +1,385 @@
+  desired_count                      = 10
+  deployment_minimum_healthy_percent = 100
+  deployment_maximum_percent         = 200
+  health_check_grace_period_seconds  = 30
+
+  load_balancer {
+    target_group_arn = aws_alb_target_group.main.arn
+    container_name   = "fleet"
+    container_port   = 8080
+  }
+
+  network_configuration {
+    subnets         = module.vpc.private_subnets
+    security_groups = [aws_security_group.backend.id]
+  }
+
+  depends_on = [aws_alb_listener.http, aws_alb_listener.https-fleetdm]
+}
+
+resource "aws_cloudwatch_log_group" "backend" {
+  name              = "fleetdm"
+  retention_in_days = 1
+}
+
+data "aws_region" "current" {}
+
+data "aws_secretsmanager_secret" "license" {
+  name = "/fleet/license"
+}
+
+resource "aws_ecs_task_definition" "backend" {
+  family                   = "fleet"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  execution_role_arn       = aws_iam_role.main.arn
+  task_role_arn            = aws_iam_role.main.arn
+  cpu                      = 1024
+  memory                   = 2048
+  container_definitions = jsonencode(
+    [
+      {
+        name      = "prometheus-exporter"
+        image     = "917007347864.dkr.ecr.us-east-2.amazonaws.com/prometheus-to-cloudwatch:latest"
+        essential = false
+        logConfiguration = {
+          logDriver = "awslogs"
+          options = {
+            awslogs-group         = aws_cloudwatch_log_group.backend.name
+            awslogs-region        = data.aws_region.current.name
+            awslogs-stream-prefix = "fleet-prometheus-exporter"
+          }
+        }
+        environment = [
+          {
+            name  = "CLOUDWATCH_NAMESPACE"
+            value = "fleet-loadtest"
+          },
+          {
+            name  = "CLOUDWATCH_REGION"
+            value = "us-east-2"
+          },
+          {
+            name  = "PROMETHEUS_SCRAPE_URL"
+            value = "http://localhost:8080/metrics"
+          },
+        ],
+      },
+      {
+        name        = "fleet"
+        image       = docker_registry_image.fleet.name
+        cpu         = 1024
+        memory      = 2048
+        mountPoints = []
+        volumesFrom = []
+        essential   = true
+        portMappings = [
+          {
+            # This port is the same that the contained application also uses
+            containerPort = 8080
+            hostPort      = 8080
+            protocol      = "tcp"
+          }
+        ]
+        ulimits = [
+          {
+            softLimit = 9999,
+            hardLimit = 9999,
+            name      = "nofile"
+          }
+        ]
+        networkMode = "awsvpc"
+        logConfiguration = {
+          logDriver = "awslogs"
+          options = {
+            awslogs-group         = aws_cloudwatch_log_group.backend.name
+            awslogs-region        = data.aws_region.current.name
+            awslogs-stream-prefix = "fleet"
+          }
+        },
+        secrets = [
```

Comment:
see previous comments about secrets manager, but I suggest for loadtesting env we just omit secrets manager usage.